src/platform/user/profile/utilities/index.js

```diff
@@ -141,3 +141,78 @@ export function teardownProfileSession() {
   sessionStorage.removeItem('shouldRedirectExpiredSession');
   clearSentryLoginType();
 }
+
+export const getValidationMessageKey = (
+  suggestedAddresses,
+  validationKey,
+  addressValidationError,
+) => {
+  let validationErrorKey;
+  const singleSuggestion = suggestedAddresses.length === 1;
+  const containsBadUnitNumber =
+    suggestedAddresses.filter(
+      address =>
+        address.addressMetaData?.deliveryPointValidation ===
+        'STREET_NUMBER_VALIDATED_BUT_BAD_UNIT_NUMBER',
+    ).length > 0;
+
+  const containsMissingUnitNumber =
+    suggestedAddresses.filter(
+      address =>
+        address.addressMetaData?.deliveryPointValidation ===
+        'STREET_NUMBER_VALIDATED_BUT_MISSING_UNIT_NUMBER',
+    ).length > 0;
+
+  if (singleSuggestion && containsBadUnitNumber) {
+    validationErrorKey = validationKey
+      ? 'badUnitNumberOverride'
+      : 'badUnitNumber';
+  }
+
+  if (singleSuggestion && containsMissingUnitNumber) {
+    validationErrorKey = validationKey
+      ? 'missingUnitNumberOverride'
+      : 'missingUnitNumber';
+  }
+
+  if (
+    singleSuggestion &&
+    !addressValidationError &&
+    !containsMissingUnitNumber &&
+    !containsBadUnitNumber
+  ) {
+    validationErrorKey = validationKey
+      ? 'showSuggestionsOverride'
+      : 'showSuggestions';
+  }
+
+  if (!singleSuggestion && !addressValidationError) {
```

Comment:
I'm sure the body of this function could be rewritten to nest the conditionals that are repeated, but I actually think it's easier to read it this way. But I think the readability would be even better if we had an additional boolean, `multipleSuggestions`, to be crystal clear about what `!singleSuggestion` really means. Someone could read `!singleSuggestion` as meaning there are _no_ suggestions at all, but that will never actually be the case